src/filters/RobotFilter.cpp

```diff
@@ -0,0 +1,185 @@
+//
+// Created by rolf on 05-11-19.
+//
+
+#include "filters/RobotFilter.h"
+#include "Scaling.h"
+
+RobotFilter::RobotFilter(const proto::SSL_DetectionRobot &detectionRobot, double detectTime, int cameraID) :
+        CameraFilter(detectTime, cameraID),
+        botId{static_cast<int>(detectionRobot.robot_id())} {
+    KalmanInit(detectionRobot);
+}
+
+bool compareObservation(const RobotFilter::RobotObservation &a, const RobotFilter::RobotObservation &b) {
+    return (a.time < b.time);
+}
+void RobotFilter::update(double time, bool doLastPredict) {
```

Comment:
naming convention. doLastPredict is called extrapolateLastStep in WorldFilter.cpp